allensdk/test/api/test_cache.py

```diff
@@ -160,33 +161,70 @@ def test_wrap_dataframe(ju_read_url_get, ju_write, mock_read_json, rma, cache):
     ju_write.assert_called_once_with('example.txt', _msg)
     mock_read_json.assert_called_once_with('example.txt', orient='records')
 
-def test_memoize():
-
-        import time
 
+def test_memoize_with_function():
+    @memoize
+    def f(x):
+        time.sleep(0.1)
+        return x
+    
+    # Build cache
+    for i in range(3):
+        uncached_result = f(i)
+        assert uncached_result == i
+    assert f.cache_size() == 3
+    
+    # Test cache was accessed
+    for i in range(3):
+        t0 = time.time()
+        result = f(i)
+        t1 = time.time()
+        assert result == i
+        assert t1 - t0 < 0.1
+    
+    # Test cache clear
+    f.clear_cache()
+    assert f.cache_size() == 0
+
+
+def test_memoize_with_kwarg_function():
+    @memoize
+    def f(x, *, y, z=1):
+        time.sleep(0.1)
+        return (x * y * z)
+    
+    # Build cache
+    f(2, y=1, z=2)
+    assert f.cache_size() == 1
+
+    # Test cache was accessed
+    t0 = time.time()
+    result = f(2, y=1, z=2)
+    t1 = time.time()
+    assert result == 4
+    assert t1 - t0 < 1.0 
```

Comment:
I think you meant for this to be `assert t1 - t0 < 0.1` since your kwarg function `f` only sleeps for `0.1` seconds?

Reply:
Yes I did, thanks. Originally I had it longer but there was no point in sleeping so long.